### PR TITLE
Log monitor results

### PR DESCRIPTION
**Pull request by shravnk**
- create new appender for MonitoringService only
- log results each time all Monitors are run

### CODE REVIEW

File: grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy [hunk 1]
```diff
@@ -156,6 +157,21 @@ class MonitoringService extends BaseService implements AsyncSupport, EventPublis
         new MonitorStatusReport(results: results)
     }
 
+    private void logResults() {
+        results.each{
```
Comment (amcclain) [inline grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy]: You could use `results.each{code, result -> ... }` here - this is often handy when iterating over maps in Groovy where you want to get nice variable names for both the key and the value in your closure (vs. getting the default `it` as a map entry).

File: grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy [hunk 2]
```diff
@@ -156,6 +157,21 @@ class MonitoringService extends BaseService implements AsyncSupport, EventPublis
...
+            def result = it.value,
+                code = result.code,
+                status = result.status.toString(),
```
Comment (amcclain) [inline grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy]: You can skip the `toString()` as all you are doing with the status is putting it in a gstring `"${status}" which will call toString() on its own.  (But good to be conscious of this generally, as passing a non-string _directly_ to the logger will throw.)

File: grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy [hunk 3]
```diff
@@ -156,6 +157,21 @@ class MonitoringService extends BaseService implements AsyncSupport, EventPublis
...
+                metric = result.metric ?: null
```
Comment (amcclain) [inline grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy]: Skip this `?: null` here - we actively don't want to take a metric of 0 -> null (always want to be careful with falsey evaluations like this against numbers), and there's no problem with letting null flow into the gstring.

**Comment by shravnk** [comment]
Thank you, made all of the above changes. Definitely agree on the additional monitor config.

Otherwise, day rolling policy works fine in the one example I have. Also apps without any monitors log only the following, which seems reasonable:
`{timestamp} | fails=0 | warns=0 | okays=0`

File: grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy [hunk 4]
```diff
@@ -156,6 +157,21 @@ class MonitoringService extends BaseService implements AsyncSupport, EventPublis
         new MonitorStatusReport(results: results)
     }
 
+    private void logResults() {
```
Comment (lbwexler) [inline grails-app/services/io/xh/hoist/monitor/MonitoringService.groovy]: need space before all of these curly braces where taking closure

File: src/main/groovy/io/xh/hoist/log/LogUtils.groovy [hunk 5]
```diff
@@ -81,10 +82,37 @@ class LogUtils {
...
+            }
+        }
+    }
+
     static void initConfig(Script script) {
         withDelegate(script) {
 
-            def appLogName = Utils.appCode
+            def appLogName = Utils.appCode,
+                    monitorLogName = "${appLogName}-monitor"
```
Comment (lbwexler) [inline src/main/groovy/io/xh/hoist/log/LogUtils.groovy]: formattting here.

File: src/main/groovy/io/xh/hoist/log/LogUtils.groovy [hunk 6]
```diff
@@ -94,6 +122,8 @@ class LogUtils {
             }
 
             dailyLog(name: appLogName, script: script)
+            monitorLog(name: monitorLogName, script: script)
+            monitorConsole(name: "monitor-console", script: script)
```
Comment (lbwexler) [inline src/main/groovy/io/xh/hoist/log/LogUtils.groovy]: prefer single quotes in groovy (like javaScript)

**Pull request merged**